ports/esp32/machine_wdt.c

```diff
@@ -42,19 +43,40 @@ typedef struct _machine_wdt_obj_t {
 STATIC machine_wdt_obj_t wdt_default = {{&machine_wdt_type}};
 
 STATIC mp_obj_t machine_wdt_make_new(const mp_obj_type_t *type_in, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 0, 1, false);
+    mp_arg_check_num(n_args, n_kw, 0, 2, true);
+    
+    enum {
+        ARG_timeout, ARG_panic
+    };
 
-    mp_int_t id = 0;
-    if (n_args > 0) {
-        id = mp_obj_get_int(args[0]);
-    }
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_timeout, MP_ARG_INT, {.u_int = 0} },
+        { MP_QSTR_panic, MP_ARG_BOOL, {.u_bool = true} }
```

Comment:
Is the current default behaviour to panic or to reset? As far as I can tell it currently resets so you should probably keep that behaviour. Doing a panic is less useful when running MicroPython.

Reply:
Hi @nickovs,
thanks for the help!
Reset is the current default behaviour so i keep it as you suggest. I have also changed default timeout from 0 to 10s to avoid a immediately reset on WDT constructor call if no timeout arg is passed.